r/adbcdrivermanager/src/error.cc

```diff
@@ -48,9 +48,28 @@ extern "C" SEXP RAdbcAllocateError(SEXP shelter_sexp) {
   return error_xptr;
 }
 
+static SEXP wrap_error_details(AdbcError* error) {
+  int n_details = AdbcErrorGetDetailCount(error);
+  SEXP result_names = PROTECT(Rf_allocVector(STRSXP, n_details));
+  SEXP result = PROTECT(Rf_allocVector(VECSXP, n_details));
+
+  for (int i = 0; i < n_details; i++) {
```

Comment:
From a C++ style standpoint I tend to use ++i here since i++ returns a value and ++i won't.  Sometimes this can equate into a performance improvement but I doubt we would see it here.